File: OneDrive/Documents/PhaseGateOne/JavaScript/CreditCardValidator.js
const prompt = require("prompt-sync")()

function cardValidity(cardNumber){
	let cardValid = false;
		if (cardNumber.charAt(0) > '2' && cardNumber.charAt(0) < '7'){
			if (cardNumber.length >= 13 && cardNumber.length <= 16){
				if (cardNumber.charAt(0) == '3' && cardNumber.charAt(1) != '7'){
				cardValid = false;
				}else {
				cardValid = true;
				};
			}else {
				cardValid = false;
				};
		} else {
			cardValid = false;
			};
	let validity = (cardValid) ? "Valid" : "Invalid";
	return validity;
	};

function cardNumberVerifier(myCardNumber){
	let myNumber = myCardNumber.replaceAll(" ", "");

	if (myNumber.length <13 || myNumber.length > 16) myNumber = "Invalid";
	for (let index = 0; index < myNumber.length; index++){
		if (myNumber.charAt(index) < '0' || myNumber.charAt(index) > '9' &&  myNumber.charAt(index) != ' '){
			myNumber = "Invalid";
		} 
		if (myNumber == "Invalid") break;
	}
	return myNumber
	};

function cardNumberLength(myCardNumber){
	let cardNumber = myCardNumber.replaceAll(" ", "");

	let digitLength = cardNumber.length;

	return digitLength;
	};
	
function creditCardType(myCardNumber){
	let cardType = "";
	let cardNumber = myCardNumber.replaceAll(" ", "");

	for (let index = 0; index < cardNumber.length; index++){
		if (cardNumber.charAt(index) < '0' || cardNumber.charAt(index) > '9' &&  cardNumber.charAt(index) != ' '){
			cardType = "Invalid";
			break;
			}
		}
	if (cardNumber.length >= 13 && cardNumber.length <= 16 && cardType != "Invalid"){
		if (cardNumber.charAt(0) == '4'){
			cardType = "VisaCard";
		} else if (cardNumber.charAt(0) == '5'){
			cardType = "MasterCard";
		} else if (cardNumber.charAt(0) == '6'){
			cardType = "DiscoverCard";
		} else if (cardNumber.charAt(0) == '3' && cardNumber.charAt(1) == '7'){
			cardType = "AmericanExpressCard";
		} else { cardType = "Invalid Card"; };
	} else { 
		cardType = "Invalid Card";
	};
	return cardType;
	};

function validityStatus(myCardNumber){
	let sum = 0;  let sumOdd = 0;
	let cardNumber = myCardNumber.replaceAll(" ", "");

	for (let index = cardNumber.length - 1; index >= 0; index--){
		sumOdd += Number(cardNumber.charAt(index));
	
		if (index > 0){ index -= 1; } else { break; };
		let integerValue = Number(cardNumber.charAt(index)) * 2;
			if (integerValue > 9){
				let firstDigit = Math.floor(integerValue / 10);  let secondDigit =  integerValue % 10;
				sum += (firstDigit + secondDigit);	
			} else{
				sum += integerValue;
			};
		};
	let total = sum + sumOdd;
	let validator = (total % 10 == 0) ? "Valid" : "Invalid";
	return validator;
	};



myCardNumber = prompt("Enter card number: ");

let stars = "";
for (let index = 0; index < 42; index++){  stars += "*"};
	console.log(stars)
	console.log("** Credit Card Type: ", creditCardType(myCardNumber));
	console.log("** Credit Card Number: ", cardNumberVerifier(myCardNumber));
	console.log("** Credit Card Digit Length: ", cardNumberLength(myCardNumber));
	console.log("** Credit Card Validity Status: ", validityStatus(myCardNumber));
	console.log(stars)

File: OneDrive/Documents/PhaseGateOne/Python/credit_card_validator.py
def cardValidity(myCardNumber):
	cardNumber = myCardNumber.replace(" ", "");
	cardValid = false
	if cardNumber[0] > '2' & cardNumber[0] < '7':
		if len(cardNumber) >= '13' & len(cardNumber) <= '16':
			if cardNumber[0] == '3' & cardNumber[1] != '7':
				cardValid = false;
			else:
				cardValid = true;
		else:	
			cardValid = false;
	else:
		cardValid = false;
	validity = ""
	if cardValid == True:
		validity = "Valid"
	else:
		validity = "Invalid"
	return validity;

def cardNumberVerifier(myCardNumber):
	myNumber = myCardNumber.replace(" ", "");
	return myNumber

def cardNumberLength(myCardNumber):
	cardNumber = myCardNumber.replace(" ", "")

	digitLength = len(cardNumber)

	return digitLength;


def creditCardType(myCardNumber):
	cardType = "";
	cardNumber = myCardNumber.replace(" ", "");

	if len(cardNumber) >= 13 & len(cardNumber) <= 16:
		if cardNumber[0] == '4':
			cardType = "VisaCard";
		elif cardNumber[0] == '5':
			cardType = "MasterCard"
		elif cardNumber[0] == '6':
			cardType = "DiscoverCard"
		elif cardNumber[0] == '3' & cardNumber[1] == '7':
			cardType = "AmericanExpressCard";
		else:
			cardType = "Invalid Card"
	else:
		cardType = "Invalid Card"
	return cardType;


def validityStatus(myCardNumber):
	sum = 0;  
	sumOdd = 0;
	cardNumber = myCardNumber.replace(" ", "");
	
	for index in range(len(cardNumber)-1, -1, -2):
		sumOdd += int(cardNumber[index]);
	for index in range(len(cardNumber)-2, -1, -2):
		integerValue = int(cardNumber[index]) * 2;
		if integerValue > 9:
			firstDigit = integerValue // 10
			secondDigit =  integerValue % 10;
			sum += (firstDigit + secondDigit)
		else:
			sum += integerValue;
	total = sum + sumOdd;
	validator = ""
	if total % 10 == 0:
		validator = "Valid"
	else:
		validator = "Invalid"
	return validator



myCardNumber = input("Enter card number: ");

stars = "";
for index in range(42):
	stars += "*"
print(stars)
print("** Credit Card Type: ", creditCardType(myCardNumber))
print("** Credit Card Number: ", cardNumberVerifier(myCardNumber))
print("** Credit Card Digit Length: ", cardNumberLength(myCardNumber))
print("** Credit Card Validity Status: ", validityStatus(myCardNumber))
print(stars)






